rosa: init at 1.2.35

jfchevrette opened the pull request:
## Description of changes

Add [rosa](https://github.com/openshift/rosa) utility to nixpkgs

[ROSA](https://docs.openshift.com/rosa/welcome/index.html) is platform for Red Hat OpenShift Service running on AWS and this is the CLI for it.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jfchevrette committed:
rosa: init at 1.2.35

AndersonTorres commented on pkgs/applications/networking/cluster/rosa/default.nix:
```diff
@@ -0,0 +1,50 @@
+    owner = "openshift";
+    repo = "rosa";
+    rev = "v${version}";
+    sha256 = "sha256-TqUtORKH+Iv8mSJikI0GZ3MvTDsNm635x0mX6iKcDFQ=";
```
```suggestion
    hash = "sha256-TqUtORKH+Iv8mSJikI0GZ3MvTDsNm635x0mX6iKcDFQ=";
```

AndersonTorres commented on pkgs/applications/networking/cluster/rosa/default.nix:
```diff
@@ -0,0 +1,50 @@
+  # project has vendored deps
+  vendorHash = null;
+
+  # Strip the final binary.
```
```suggestion
```

jfchevrette replied:
removed in c6751b0

AndersonTorres commented on pkgs/applications/networking/cluster/rosa/default.nix:
```diff
@@ -0,0 +1,50 @@
+    platforms = platforms.all;
+  };
+}
+
```
```suggestion
```

jfchevrette replied:
removed in c6751b0

jfchevrette commented:
Thanks for the review @AndersonTorres 

In c6751b0 I removed comments and moved attributes around a bit so related attributes are together (to be best of my knowledge). I also removed `doCheck = !stdenv.isDarwin` in favor of `__darwinAllowLocalNetworking` so that tests will work on darwin when sandbox is enabled

jfchevrette commented:
I have a PR upstream (https://github.com/openshift/rosa/pull/1691) to add an option for the version check to be local only, until then I have removed the version check test

AndersonTorres commented on pkgs/applications/networking/cluster/rosa/default.nix:
```diff
@@ -0,0 +1,36 @@
+    license = licenses.asl20;
+    homepage = "https://github.com/openshift/rosa";
+    maintainers = with maintainers; [ jfchevrette ];
+    platforms = platforms.all;
```
```suggestion
```

Go does not exist on all platforms.

jfchevrette replied:
Changed to use `go.meta.platforms`

AndersonTorres commented on pkgs/top-level/all-packages.nix:
```diff
@@ -35066,6 +35066,8 @@ with pkgs;
 
   rootlesskit = callPackage ../tools/virtualization/rootlesskit { };
 
+  rosa = callPackage ../applications/networking/cluster/rosa { };
```
Move it to the pkgs/by-name hierarchy

jfchevrette replied:
done. since this is fairly new I didn't know this existed. I also found the [docs](https://github.com/NixOS/nixpkgs/commits/master/pkgs/by-name/README.md) about it. Thanks for your patience and guidance

jfchevrette commented:
bumped version to 1.2.35 + rebased + squashed commits

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3527

AndersonTorres reviewed (commented):
LGTM. Waiting for other reviewers.

SuperSandro2000 commented on pkgs/by-name/ro/rosa/package.nix:
```diff
@@ -0,0 +1,43 @@
+    license = licenses.asl20;
+    homepage = "https://github.com/openshift/rosa";
+    maintainers = with maintainers; [ jfchevrette ];
+    platforms = go.meta.platforms;
```
```suggestion
```
that is already being set

jfchevrette replied:
I didn't know `buildGoModule` already set that. Thank you for pointing it out!

SuperSandro2000 commented on pkgs/by-name/ro/rosa/package.nix:
```diff
@@ -0,0 +1,43 @@
+      --zsh <($out/bin/rosa completion zsh)
+  '';
+
+  preCheck = ''
+    ln -s $GOPATH/bin/rosa rosa
+  '';
```
How does that interact with go test?

jfchevrette replied:
That is something I copied from a related package 
https://github.com/NixOS/nixpkgs/blob/95002f7df788ca7b89fb16cd5241345c33839035/pkgs/applications/networking/cluster/ocm/default.nix#L22-L24
without really knowing why it was needed or even if this new package needed it. Looks like it does not. Lesson learned :)
removed in 1839dd30e1d5

jfchevrette commented:
Result of `nixpkgs-review pr 279891` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rosa</li>
  </ul>
</details>

AndersonTorres reviewed (changes requested):
Now squash the commits.

Besides LGTM.

Result of `nixpkgs-review pr 279891` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rosa</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3631

SuperSandro2000 commented on pkgs/by-name/ro/rosa/package.nix:
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles, stdenv, testers, rosa, go }:
```
```suggestion
{ lib, buildGoModule, fetchFromGitHub, installShellFiles, testers, rosa }:
```

jfchevrette replied:
fixed. thank you!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3755
